src/networking/backend.ts

```diff
@@ -104,6 +107,10 @@ export class Backend {
       supports_direct_payment: true,
     };
 
+    if (purchaseMetadata) {
+      requestBody.purchase_metadata = purchaseMetadata;
```

Comment:
For the existing metadata in the POST /receipt endpoint in iOS, we accept a `metadata` field, of type `Record<string, string>`. So we only accept string values: https://github.com/RevenueCat/purchases-ios/blob/15c67454493cd0a483e2a3f445d7216675462893/Sources/Networking/Operations/PostReceiptDataOperation.swift#L144

I'm not sure if we want to reuse the same behavior, or have an entirely different mechanism for purchases-js... Note that any differences might make it more difficult to integrate with the hybrids later + we need to make sure the data matches the behavior in the backend, so I would try to consolidate the behavior unless there is a reason to keep it separate.

Reply:
Renamed it into `metadata`